File: tests/nodejs/cases/case_218_thread_pool_capture_02.js
/**
 * Task case 218: thread_pool_capture_02 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case218ThreadPoolCapture02(input) {
  const taskName = 'thread_pool_capture_02';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: async task captures payload.
  setTimeout(() => retainedAudit.push(payload), 0);
  return 'ok';
}

module.exports = { case218ThreadPoolCapture02 };

File: tests/go/cases/case_218_thread_pool_capture_02.go
package escape_tests

var retainedCase218 = []map[string]string{}

func Case218ThreadPoolCapture02(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "thread_pool_capture_02",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "thread_pool_capture_02:" + raw,
	}
	// ESCAPE: goroutine captures payload and retains it.
	go func(v map[string]string) { retainedCase218 = append(retainedCase218, v) }(payload)
	return "ok"
}
